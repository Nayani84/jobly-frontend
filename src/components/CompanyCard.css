/* General JobCard container styling */
.CompanyCard {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    border: 1px solid #ddd;
    border-radius: 8px;
    padding: 10px;
    margin: 10px 0;
    background-color: #f9f9f9;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
    transition: box-shadow 0.3s ease;
}

.CompanyCard:hover {
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.2);
}

.CompanyCard h3 {
    font-size: 1.5rem;
    margin: 8px 0px;
    color: #333;
}

.CompanyCard p {
    margin: 4px 0;
    font-size: 1rem;
    color: #555;
}