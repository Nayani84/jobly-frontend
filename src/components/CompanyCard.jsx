import React from "react";
import { Link } from "react-router-dom";
import "./CompanyCard.css";

function CompanyCard({ company }) {

    return (
        <div className="CompanyCard">

            <h3><Link to={`/companies/${company.handle}`}>{company.name}</Link></h3>
            <p>{company.description}</p>

        </div>
    );
}

export default CompanyCard;