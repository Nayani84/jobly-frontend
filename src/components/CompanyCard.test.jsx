import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyCard from "./CompanyCard";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <CompanyCard company={{ handle: "test-company", name: "Test Company", description: "This is a test company." }} />
        </MemoryRouter>
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <CompanyCard company={{ handle: "test-company", name: "Test Company", description: "This is a test company." }} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("displays company details correctly", function () {
    const { getByText } = render(
        <MemoryRouter>
            <CompanyCard company={{ handle: "test-company", name: "Test Company", description: "This is a test company." }} />
        </MemoryRouter>
    );

    expect(getByText("Test Company")).toBeInTheDocument();
    expect(getByText("This is a test company.")).toBeInTheDocument();
});
