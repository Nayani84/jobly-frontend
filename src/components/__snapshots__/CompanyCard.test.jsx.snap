// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CompanyCard"
  >
    <h3>
      <a
        data-discover="true"
        href="/companies/test-company"
      >
        Test Company
      </a>
    </h3>
    <p>
      This is a test company.
    </p>
  </div>
</DocumentFragment>
`;
